cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -74,7 +73,7 @@ func GetMetricStorageFactory(name string, host component.Host) (storage.MetricSt
 	return mf, nil
 }
 
-func GetStorageFactoryV2(name string, host component.Host) (tracestore.Factory, error) {
+func GetStorageFactoryV2(name string, host component.Host) (*factoryadapter.Factory, error) {
```

Comment:
don't know about this - adapter is implementation detail, this function should not leak it

Reply:
@yurishkuro I was trying to follow's Go's best practice here of returning structs instead of interfaces. If we want to keep the interface return, we'll need to define another function `GetDependencyStoreFactory` with a different return type (`depstore.Reader`) but it will do the exact same thing. Let me know what you think.

Reply:
it's not the best practice in this case. The return-struct best practice applies to constructors. `GetStorageFactoryV2` is not a constructor, it's an accessor that is supposed to return a storage interface. The caller of this function is not supposed to know about the implementation of that interface and accordingly should not be able to take dependency on the concrete returned type, which could be wider than the interface alone.

Reply:
Sounds good - so should we have two accessors here (something like `GetTraceStoreFactory` and `GetDependencyStoreFactory`)? Their return types would be different but the implementations would be the same.

Reply:
yes, we should. Implementations are only the same _today_, not in the future.

Reply:
sounds good to me! done